src/ucp/core/ucp_worker.c

```diff
@@ -750,18 +753,152 @@ void ucp_worker_iface_event(int fd, void *arg)
     ucp_worker_signal_internal(worker);
 }
 
+static int ucp_worker_iface_find_better(ucp_worker_h worker,
+                                        ucp_worker_iface_t *wiface)
+{
+    ucp_context_h ctx = worker->context;
+    ucp_rsc_index_t id;
+    ucp_worker_iface_t *if_iter;
+
+    for (id = 0; id < ctx->num_tls; ++id) {
+        if_iter = &worker->ifaces[id];
+
+        /* Need to check resources which belong to the same device only */
+        if ((ctx->tl_rscs[id].dev_index != ctx->tl_rscs[wiface->rsc_index].dev_index) ||
+            (if_iter->flags & UCP_WORKER_IFACE_FLAG_SUBOPTIMAL) ||
+            (id == wiface->rsc_index)) {
+            continue;
+        }
+
+        /* Check that another iface:
+         * 1. Supports all capabilities of the target iface (at least)
+         * 2. Has the same or better performance charasteristics */
+        if (ucs_test_all_flags(if_iter->attr.cap.flags, wiface->attr.cap.flags) &&
```

Comment:
do we really need to test all flags? maybe define some mask of what is really needed?

Reply:
IMO, better check that all caps are covered. Flags can be relevant to one of the features (maybe except UCT_IFACE_FLAG_CONNECT_TO_IFACE). We could try to use different masks depending on features requested, but what would be the real use case which benefits from that?